eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/io/AccessionReportWriter.java

```diff
@@ -117,31 +178,81 @@ public void update(ExecutionContext executionContext) throws ItemStreamException
+            variantsWriter.close();
+            if (!duplicatedInputContigsToInsdc.isEmpty()) {
+                logger.warn(
+                        "The same chromosome (in the original input) was replaced by several INSDC contig accessions. "
+                        + "This happened for: " + duplicatedInputContigsToInsdc.toString());
+            }
+            if (!duplicatedInsdcToInputContigs.isEmpty()) {
+                logger.warn("The same INSDC contig accessions replaced several input chromosomes. This happened for: "
+                            + duplicatedInsdcToInputContigs.toString());
+            }
         } catch (IOException e) {
             throw new ItemStreamException(e);
         }
     }
 
-    public void write(List<? extends AccessionWrapper<ISubmittedVariant, String, Long>> accessions,
-                      AccessionWrapperComparator accessionWrapperComparator) throws IOException {
-        if (fileWriter == null) {
-            throw new IOException("The file " + output + " was not opened properly. Hint: Check that the code " +
-                                          "called AccessionReportWriter::open");
+    public void write(List<? extends IVariant> originalVariantsWithInsdcContigs,
+                      List<AccessionWrapper<ISubmittedVariant, String, Long>> accessionedVariants) throws IOException {
+        if (variantsWriter == null) {
+            throw new IOException("The file " + variantsOutput + " was not opened properly. Hint: Check that the code "
+                                  + "called " + this.getClass().getSimpleName() + "::open");
+        }
+        updateChromosomeMappings(originalVariantsWithInsdcContigs);
+        List<? extends AccessionWrapper<ISubmittedVariant, String, Long>> denormalizedVariants = denormalizeVariants(
+                accessionedVariants);
+        denormalizedVariants.sort(new AccessionWrapperComparator(originalVariantsWithInsdcContigs));
+        for (AccessionWrapper<ISubmittedVariant, String, Long> variant : denormalizedVariants) {
+            writeSortedVariant(variant, insdcToInputContigs);
         }
-        List<? extends AccessionWrapper<ISubmittedVariant, String, Long>> denormalizedAccessions = denormalizeVariants(
-                accessions);
-        denormalizedAccessions.sort(accessionWrapperComparator);
-        for (AccessionWrapper<ISubmittedVariant, String, Long> variant : denormalizedAccessions) {
-            writeVariant(variant.getAccession(), variant.getData());
+        variantsWriter.flush();
+    }
+
+    private void updateChromosomeMappings(List<? extends IVariant> originalVariantsWithReplacedContigs)
+            throws IOException {
+        for (IVariant variantWithContig : originalVariantsWithReplacedContigs) {
+            String originalChromosome = getOriginalChromosome(variantWithContig);
+
+            String previousContig = inputContigsToInsdc.put(originalChromosome, variantWithContig.getChromosome());
```

Comment:
Suggest collecting variantWithContig.getChromosome() in a variable currentContig and use that subsequently. This makes the rest of the code intuitive.